sql/testdata/timedate

```diff
@@ -0,0 +1,38 @@
+statement ok
+CREATE TABLE t (
+  a TIMESTAMP PRIMARY KEY,
+  b DATE,
+  c INTERVAL,
+  CONSTRAINT bidx UNIQUE (b),
+  CONSTRAINT cidx UNIQUE (c)
+)
+
+# TODO(vivek) replace type casting in this file.
```

Comment:
Replace with what? The casting shouldn't be necessary with `convertDatum` parsing strings, correct?